CONTRIBUTING.md

```diff
@@ -29,34 +29,142 @@ This project follows
+   and well-maintained projects?
 1. What version of the package should we depend on?
-1. Are the licenses for the package and its transitive dependencies green? See next section for checking licenses.
+1. Are the licenses for the package and its transitive dependencies green? See
+   next section for checking licenses.
 
 ### Checking NPM dependencies licenses
 
-When adding any new dependencies or updating existing dependencies
-we need to check licenses:
+When adding any new dependencies or updating existing dependencies we need to
+check licenses:
 
 ```console
 npm run lint-license
 ```
 
+### Local development
+
+When developing locally, you often want to use the the local modified version of
+NPM packages (`kpt-functions` and `create-kpt-functions`) used by function
+packages (e.g. `demo-functions`). You can do the following:
+
+```console
+cd ts/demo-functions
+# May need to run as `sudo` depending on how you intalled NPM
+npm link ../kpt-functions
+npm run build
+```
+
 ### Releases
 
-Release workflows are triggered by creating Releases in the GitHub UI with a tag following the
-format:
+This repo contains two NPM packages that are published to the NPM registry.
+
+#### `kpt-functions`
+
+This is the library used by all TS SDK functions.
+
+##### NPM registry
+
+https://www.npmjs.com/package/kpt-functions
+
+##### Release candidates
+
+Before releasing major changes, you first want to create a `rc` release.
+
+1.  In pristine git repo, run:
+
+    ```console
+    ./scripts/version-kpt-functions-sdk.sh 0.14.0-rc.1
+    ```
+
+    This automatically creates a Git commit in your local repo.
+
+2.  Create a PR and commit.
+    [Example](https://github.com/GoogleContainerTools/kpt-functions-sdk/commit/d944c818f564a183c3cb092b282f5e83f770b18a)
+3.  Create a release in GitHub which pushes the package to NPM registry. But,
+    because this is marked as an `rc` release, it will not be pulled by default
+    form the NPM registry.
+    [Example](https://github.com/GoogleContainerTools/kpt-functions-sdk/releases/tag/release-kpt-functions-v0.14.0)
+
+4.  In a separate PR, update the dependant packages
+
+    ```console
+    ./scripts/version-kpt-functions-sdk-deps.sh 0.14.0-rc.1
+    ```
+
+5.  Create a PR and commit.
+    [Example](https://github.com/GoogleContainerTools/kpt-functions-sdk/commit/e1126e5a23fac3d3a79706ceaca924a9b4d31a18)
+    This PR ensures that dependant packages are tested against the `rc` release.
+
+6.  Once you are confident that the `rc` release is good, you can then repeat
+    the process with out the `rc` suffix:
+    ```console
+    ./scripts/version-kpt-functions-sdk.sh 0.14.0
+    ```
+7.  In a separate PR, update the SDK API docs which are
+    [hosted here](https://googlecontainertools.github.io/kpt-functions-sdk/api/)
+
+    ```console
+    cd ts/kpt-functions
+    npm run gen-docs
+    ```
+
+    **NOTE**: You want to merge this PR. Do not squash or rebase this PR. The
```

Comment:
The repo setting currently only allow `squash and merge` and `rebase and merge`. We should change the setting to achieve it.

Reply:
Enable merge commits